### DISCUSSION TITLE
Comparison and Roadmap: `WarpX RZ` vs `FBPIC`

### DISCUSSION THREAD

**Discussion by berceanu** [BLAST-WarpX/warpx#4384] [Q&A] [answered]
It seems there is significant overlap in functionality between `WarpX`'s RZ mode with `USE_PSATD=TRUE` and `FBPIC`.

What are the main differences and when do you recommend using one or the other?
What is the development roadmap going forward for the two codes?

**Comment by RemiLehe** [comment, marked as answer]
Sorry for the late reply!

Yes, you are correct that the algorithm in WarpX's RZ PSATD mode is the same as FBPIC.

However, there are still some small differences between these two. The most significant ones are probably:
- FBPIC has a modified algorithm for charge/current deposition close to the axis (https://github.com/fbpic/fbpic/pull/347), which has not been ported yet to WarpX, and is in principle more accurate for intense beams concentrated near the axis.
- WarpX can run on AMD and NVIDIA GPUs, while FBPIC can only run on NVIDIA GPUs at this point.
- The default number of guard cells is more conservative in FBPIC, and this makes simulations maybe more robust for users that don't change this default value.

At this point, the WarpX development team is much larger than the FBPIC team, and therefore development is more rapid in WarpX. However, we continue to maintain and improve FBPIC, since a significant part of the laser-plasma community is currently using it. (One example is the recent support for `lasy` files in FBPIC.)
